### PR TITLE
#895: Migrate to Junit 5

### PR DESCRIPTION
**Pull request by andreoss**
Per #895 

- Migrate to Junit 5

### CODE REVIEW

File: src/test/java/org/takes/facets/auth/social/PsGoogleTest.java [hunk 1]
```diff
@@ -211,50 +211,56 @@ public void exec(final URI home) throws IOException {
...
+    void badGoogleResponse() {
+        Assertions.assertThrows(
+            IOException.class,
+            () -> {
+                final Take take = new TkFork(
+                    this.requestToken(),
+                    new FkRegex(
+                        PsGoogleTest.REGEX_PATTERN,
+                        // @checkstyle AnonInnerLengthCheck (1 line)
+                        new Take() {
```
Comment (baudoliver7) [inline src/test/java/org/takes/facets/auth/social/PsGoogleTest.java:225]: @andreoss Could you use a lambda expression ?

File: src/test/java/org/takes/misc/VerboseListTest.java [hunk 2]
```diff
@@ -55,6 +56,7 @@
      * Rule for asserting thrown exceptions.
      */
     @Rule
+    @SuppressWarnings("deprecation")
     public final ExpectedException thrown = ExpectedException.none();
```
Comment (baudoliver7) [inline src/test/java/org/takes/misc/VerboseListTest.java]: @andreoss Could you use here `Assertions.assertThrows` ?

File: src/test/java/org/takes/rs/RsPrintTest.java [hunk 3]
```diff
@@ -38,19 +39,23 @@
...
      */
-    @Test(expected = IllegalArgumentException.class)
-    public void failsOnInvalidHeader() throws IOException {
-        new RsPrint(new RsWithHeader("name", "\n\n\n")).asString();
+    @Test
+    void failsOnInvalidHeader() {
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
+            () -> {
+                new RsPrint(new RsWithHeader("name", "\n\n\n")).asString();
```
Comment (baudoliver7) [inline src/test/java/org/takes/rs/RsPrintTest.java:52]: @andreoss I think you don't need `{}` here.

File: src/test/java/org/takes/tk/TkSlf4jTest.java [hunk 4]
```diff
@@ -25,48 +25,57 @@
...
      */
-    @Test(expected = IOException.class)
-    public void logsException() throws Exception {
-        new TkSlf4j(new TkFailure(new IOException(""))).act(new RqFake());
+    @Test
+    void logsException() {
+        Assertions.assertThrows(
+            IOException.class,
+            () -> {
+                new TkSlf4j(new TkFailure(new IOException(""))).act(new RqFake());
```
Comment (baudoliver7) [inline src/test/java/org/takes/tk/TkSlf4jTest.java:55]: @andreoss I think you don't need `{}` in lambda expression.

File: src/test/java/org/takes/tk/TkSlf4jTest.java [hunk 5]
```diff
@@ -25,48 +25,57 @@
...
      */
-    @Test(expected = RuntimeException.class)
-    public void logsRuntimeException() throws Exception {
-        new TkSlf4j(new TkFailure(new RuntimeException(""))).act(new RqFake());
+    @Test
+    void logsRuntimeException() {
+        Assertions.assertThrows(
+            RuntimeException.class,
+            () -> {
+                new TkSlf4j(new TkFailure(new RuntimeException(""))).act(new RqFake());
```
Comment (baudoliver7) [inline src/test/java/org/takes/tk/TkSlf4jTest.java:68]: @andreoss Ditto

**Review by baudoliver7 — CHANGES_REQUESTED** [review]
@andreoss Please, see my comments.

**Comment by andreoss** [comment]
@baudoliver7 Done

**Comment by yegor256** [comment]
@andreoss there are conflicts :(

**Comment by andreoss** [comment]
@yegor256 Fixed

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**